### ISSUE TITLE
Third Party Login creates different email uniqueID@loopback.facebook.com

### ISSUE THREAD

**Issue by kashifeqbal** [strongloop/loopback-component-passport#160]
If a person using facebook and gmail for login, in both the account email address is same, but it create two different user account. And User model should have original Email address rather creating a email like this 

> "uniqueID@loopback.facebook.com".

While in file user-identity.js, there is a

```
function profileToUser(provider, profile, options) {
  // Let's create a user for that 
    var email = profile.emails && profile.emails[0] && profile.emails[0].value;
    if (!email && !options.emailOptional) {
      // Fake an e-mail
      email = (profile.username || profile.id) + '@loopback.' +
              (profile.provider || provider) + '.com';
    }
    var username = provider + '.' + (profile.username || profile.id);
    var password = utils.generateKey('password');
    var userObj = {
      username: username,
      password: password
    };
    if (email) {
      userObj.email = email;
    }
    return userObj;
  }
```

var email is taking up original email from profile object. If that email is undefined then only it's create a Fake an e-mail.

So In my project I cannot save 

> uniqueID@loopback.facebook.com

want to save original email.

**Comment by PhuongNL** [comment]
you edit this line 
`var email = provider === 'ldap' ? profileEmail : generatedEmail;`
to 
`var email = provider === 'ldap' ? profileEmail : profileEmail ? profileEmail : generatedEmail;`

**Comment by jannyHou** [comment]
@kashifeqbal I think this is the related pr: https://github.com/strongloop/loopback-component-passport/pull/113

**Comment by etsuo** [comment]
I believe I'm running into this problem too. A user logs in with Facebook, which returns an email address that matches that user's existing local account; loopback-component-passport creates a new user with the email address {facebookId}@loopback.facebook.com. 

This is documented as the expected behavior in the documentation: [here](https://github.com/strongloop/loopback-component-passport/blob/5a8d31478264a2a9fc235fd97ab8e7e5c566e8fa/README.md#all-third-party-accounts-will-login-with-an-email-of-uniqueidloopbackprovidercom-example-123456loopbackfacebookcom)

But this isn't the behavior that I would expect. I'd expect if someone creates a local account, then uses facebook, then uses google, all three user identities would be related to the same user.

Am I missing something here?

**Comment by etsuo** [comment]
@jannyHou, I can't speak for @kashifeqbal, but for me  the mod by @PhuongNL works.

Is there a plan to merge this?

**Comment by cslinmiso** [comment]
No offense, but I wonder this very feature is really what people need?

**Comment by clockworkgr** [comment]
@etsuo : Possible scenario... 

Let's say I know you have an account on a "sensitive" loopback based service..I don't know your password for that service but due to some clever social engineering or an unattended laptop at work or something, I have access to your fb password or a logged-in fb account... 

Presto, I have access to your account at the sensitive loopback service.

Second scenario: I have an account on a loopback based service with some@email.com . I want to be able to log in with my google account someemail@gmail.com. I SHOUDL be able to link those two.

In short, LB should correctly NOT link accounts simply because they provide/refer to the same e-mail. Accounts should be linked when the same person chooses to link them and logs in with both of them .

**Comment by chkuendig** [comment]
This really depends on the use case, if you have a consumer-app without much sensitive data and try to minimize any login/signup friction (e.g. mobile game, newest social whatever app) you'd want to merge. 

Best would be to make this optional...
